### PR TITLE
feat: Move to vue and @nextcloud/files actions

### PR DESCRIPTION
**Pull request by juliushaertl**
Bring compatibility with Nextcloud 28 by implementing the new file
actions API. The modal is further moved to vue js and typescript is now
used.

Fix https://github.com/nextcloud/files_zip/issues/165

### CODE REVIEW

File: webpack.js
Comment (max-nextcloud) [inline webpack.js:1]: Do we still need webpack at all?

**Review by max-nextcloud — COMMENTED** [review]
Looks good at first view. Have not tested it yet though.

I'm wondering what we still need webpack for now.

File: src/services.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		}))
+		return null
+	}
+
+	const target = await askForName(nodes)
+	if (target === null) {
+		return null
+	}
+
+	compressFiles(fileIds, dir + '/' + target)
```
Comment (luka-nextcloud) [inline src/services.ts:77]: Missing await here, `compressFiles` is an async func.

File: src/services.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+			target,
+		})
+		showSuccess(t('files_zip', 'Creating Zip archive started. We will notify you as soon as the archive is available.'))
+	} catch (e) {
+		showError(t('files_zip', 'An error happened when trying to compress the file.'))
+	}
+}
+
+export const action = async (dir: string, nodes: Node[]) => {
+	const fileIds: number[] = nodes.map(file => file.fileid)
```
Comment (luka-nextcloud) [inline src/services.ts:62]: I think it should be const fileIds: number[] = nodes.map(file => file.fileid) as number[] to bypass Typescript error.

File: src/services.ts (hunk 1 above)
Comment (juliushaertl) [reply src/services.ts]: Good catch

**Comment by juliushaertl** [comment]
All resolved

File: package-lock.json
Comment (max-nextcloud) [inline package-lock.json:1]: I think we'd still want to keep the package lock to lock the dependencies to specific versions, right?

**Review by max-nextcloud — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (juliushaertl) [reply package-lock.json:1]: Yes, it is not removed, i think the github stat indicator is just confusing here

**Review by luka-nextcloud — APPROVED** [review]

**Pull request merged**